### PR TITLE
Joining tables

### PR DESCRIPTION
**Pull request by joyapisi**
## Work Done:
- Create a table named vets
L1

- Create a "join table" called specializations to handle relationship between species and vets tables
L2

-  Create a "join table" called visits to handle relationship between animals and vets table.
L3

- Insert data into vets table.
L4

- Insert data into specialties table
insert into specialization

- Insert data into visits table
insert into visits

- Write queries to answer the following:
  - Who was the last animal seen by William Tatcher?
q1

  - How many different animals did Stephanie Mendez see?
q2

  - List all vets and their specialties, including vets with no specialties.
q3

  - List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
q4

  - What animal has the most visits to vets?
q5

  - Who was Maisy Smith's first visit?
q6

  - Details for most recent visit: animal information, vet information, and date of visit.
q7

  - How many visits were with a vet that did not specialize in that animal's species?
SL

  - What specialty should Maisy Smith consider getting? Look for the species she gets the most.
last

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -110,3 +106,81 @@ JOIN animals ON owners.id = animals.owner_id
...
+LIMIT 1;
+
+
+SELECT COUNT(*) AS num_visits
+FROM visits v
+JOIN vets vt ON v.vets_id = vt.id
+JOIN animals a ON v.animals_id = a.id
+JOIN specializations s ON vt.id = s.vets_id AND a.id = s.species_id
+WHERE s.species_id IS NULL;
+
```
Comment (nahnahdev) [inline queries.sql]: - [ ] Great job so far @joyapisi There are just a few changes you need to make and you'll be on your way to the next project :+1: 
- [ ] I noticed that the result of your last two queries was incorrect. For the first query, even from the data you inserted, Maisy Smith is the vet without a specialization. However, she had about 9 visits from different animals. But your query brought back the image below with the number of visits as 0. Which makes it wrong. So kindly fix this :+1: 
image

File: queries.sql [hunk 2]
```diff
@@ -110,3 +106,81 @@ JOIN animals ON owners.id = animals.owner_id
...
+
+SELECT a.name, COUNT(*) AS num_visits
+FROM visits v
+JOIN animals a ON v.animals_id = a.id
+JOIN vets vt ON v.vets_id = vt.id
+WHERE vt.name = 'Maisy Smith'
+GROUP BY a.name
+ORDER BY num_visits DESC
+LIMIT 1;
```
Comment (nahnahdev) [inline queries.sql]: - [ ] Lastly, this query is also supposed to return the specialty that  Maisy Smith should consider getting. You can look for the species she gets the most to determine this. But from the screenshot you provided which I have pasted below, your query is returning `Boarmon` which isn't a specialty but an animal :+1: 
- [ ] So your query should return either `Digimon` or `Pokemon` :wink: 
- Kindly fix this as well and replace those queries in your PR with the right screenshots :+1: 
image

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @joyapisi , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :fire: 
- All required files present :heavy_check_mark: 
- Good use of GitHub flow :heavy_check_mark: 
- Professional README :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mwanawabangona — APPROVED** [review]
Hi @joyapisi ,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implementing the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**